src/component/providers/AccessProvider/AccessProvider.tsx

```diff
@@ -1,88 +1,110 @@
-import { FC } from 'react';
-
-import AccessContext from '../../../contexts/AccessContext';
-import useUser from '../../../hooks/api/getters/useUser/useUser';
+import { ReactElement, ReactNode, useCallback, useMemo } from 'react';
+import AccessContext, { IAccessContext } from '../../../contexts/AccessContext';
+import { useAuth } from '../../../hooks/api/getters/useAuth/useAuth';
 import { ADMIN } from './permissions';
+import { IPermission } from '../../../interfaces/user';
 
-// TODO: Type up redux store
-interface IAccessProvider {
-    store: any;
-}
-
-interface IPermission {
-    permission: string;
-    project?: string | null;
-    environment: string | null;
+interface IAccessProviderProps {
+    children: ReactNode;
+    permissions: IPermission[];
 }
 
-const AccessProvider: FC<IAccessProvider> = ({ store, children }) => {
-    const { permissions } = useUser();
-    const isAdminHigherOrder = () => {
-        let called = false;
-        let result = false;
-
-        return () => {
-            if (called) return result;
-            const permissions = store.getState().user.get('permissions') || [];
-            result = permissions.some(
-                (p: IPermission) => p.permission === ADMIN
+const AccessProvider = (props: IAccessProviderProps): ReactElement => {
+    const { auth } = useAuth();
+    const permissions = props.permissions ?? auth?.permissions;
+
+    const isAdmin: boolean = useMemo(() => {
+        return checkAdmin(permissions);
+    }, [permissions]);
+
+    const hasAccess = useCallback(
+        (permission: string, project?: string, environment?: string) => {
+            return checkPermissions(
+                permissions,
+                permission,
+                project,
+                environment
             );
+        },
+        [permissions]
```

Comment:
Given recent discussions, perhaps we should change this to JSON.stringify(permissions) or use deep comparison hook? As far as I know useCallback also does a shallow comparison.

Reply:
The `permissions` list from `useAuth` should be a stable reference now (through `useMemo`), but a good idea to double check. :eyes: